File: Eventili/src/entities/reponse.java
package entities;

import java.sql.Timestamp;

public class reponse {

    private int id_rep;
    private Personne sender;
    private reclamation rec;
    private String message;
    private Timestamp dateHeure;
    private boolean isEdited;

    public reponse() {
    }

    public reponse(int id_rep, Personne sender, reclamation rec, Timestamp dateHeure, String message, boolean isEdited) {
        this.id_rep = id_rep;
        this.sender = sender;
        this.rec = rec;
        this.message = message;
        this.dateHeure = dateHeure;
        this.isEdited = isEdited;
    }

    public reponse(Personne sender, reclamation rec, String message) {
        this.sender = sender;
        this.rec = rec;
        this.message = message;
        this.isEdited = false;
        this.dateHeure = new Timestamp(System.currentTimeMillis());
    }

    public int getIdRep() {
        return this.id_rep;
    }

    public Personne getSender() {
        return sender;
    }

    public reclamation getRec() {
        return rec;
    }
    public boolean getIsEdited() {
        return this.isEdited;
    }

    public Timestamp getTimeStamp() {
        return this.dateHeure;
    }

    public String getMessage() {
        return this.message;
    }

    public void setIdRep(int id_rep) {
        this.id_rep = id_rep;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void setTimeStamp(Timestamp timestamp) {
        this.dateHeure = timestamp;
    }

    public void setIsEdited(boolean isEdited) {
       this.isEdited = isEdited;
    }
    public void setSender(Personne sender) {
        this.sender = sender;
    }
}

File: web/eventiliProject/src/Entity/Reponse.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

/**
 * Reponse
 *
 * @ORM\Table(name="reponse", indexes={@ORM\Index(name="fk_sender", columns={"senderId"}), @ORM\Index(name="fk_rec", columns={"rec_id"})})
 * @ORM\Entity
 */
class Reponse
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_rep", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idRep;

    /**
     * @var string
     *
     * @ORM\Column(name="message", type="text", length=65535, nullable=false)
     */
    private $message;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="timestamp", type="datetime", nullable=false, options={"default"="CURRENT_TIMESTAMP"})
     */
    private $timestamp = 'CURRENT_TIMESTAMP';

    /**
     * @var bool
     *
     * @ORM\Column(name="isEdited", type="boolean", nullable=false)
     */
    private $isedited = '0';

    /**
     * @var \Reclamation
     *
     * @ORM\ManyToOne(targetEntity="Reclamation")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="rec_id", referencedColumnName="id_rec")
     * })
     */
    private $rec;

    /**
     * @var \Personne
     *
     * @ORM\ManyToOne(targetEntity="Personne")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="senderId", referencedColumnName="id_pers")
     * })
     */
    private $senderid;

    public function getIdRep(): ?int
    {
        return $this->idRep;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): self
    {
        $this->message = $message;

        return $this;
    }

    public function getTimestamp(): ?\DateTimeInterface
    {
        return $this->timestamp;
    }

    public function setTimestamp(\DateTimeInterface $timestamp): self
    {
        $this->timestamp = $timestamp;

        return $this;
    }

    public function isIsedited(): ?bool
    {
        return $this->isedited;
    }

    public function setIsedited(bool $isedited): self
    {
        $this->isedited = $isedited;

        return $this;
    }

    public function getRec(): ?Reclamation
    {
        return $this->rec;
    }

    public function setRec(?Reclamation $rec): self
    {
        $this->rec = $rec;

        return $this;
    }

    public function getSenderid(): ?Personne
    {
        return $this->senderid;
    }

    public function setSenderid(?Personne $senderid): self
    {
        $this->senderid = $senderid;

        return $this;
    }


}
